arelle/plugin/validate/ROS/rules/__init__.py

```diff
@@ -17,16 +18,41 @@ def errorOnRequiredFact(
         message: str,
 ) -> Iterable[Validation]:
     """
-    Yields an error if a fact with the given QName is not tagged with a non-nil value.
+    Yields an error if a fact with the given localName is not tagged with a non-nil value.
     :return: Yields validation errors.
     """
     facts: set[ModelFact] = modelXbrl.factsByLocalName.get(conceptLn, set())
     for fact in facts:
         if not fact.isNil:
             return
```

Comment:
If a list of facts gets run through this we do not indicate which facts are actually in error. It seems like this should be appending fats to a list and then returning that list as the model objects of the error?

Reply:
I think the function name may be misleading. The intent is to error on a _missing_ required fact. There are no facts to include in the error, just the concept for which a fact is missing. Suggest `errorOnMissingRequiredFact`?

Reply:
I don't think that's necessarily wrong to have a fact with a nil-value , the rule seems to indicate that one fact needs to be non-nil.  Not all of them.